packages/yarnpkg-libzip/sources/libzipImpl.ts

```diff
@@ -196,7 +184,7 @@ export class LibZipImpl implements ZipImpl {
         const memory = this.libzip.HEAPU8.subarray(buffer, buffer + size);
         const data = Buffer.from(memory);
 
-        return {data, compressionMethod};
+        return {data, compressionMethod, shouldBeCached: true};
```

Comment:
I'd prefer `shouldBeCached` to be a class property than something returned by `getFileSource`, especially since it never changes (and probably doesn't have a good reason to ever change?).

Reply:
I was thinking that backend could decide whether to cache on per file basis. 
E.g for big compressed files. Or frequently requested files. You decide.

Reply:
Please reply if you want per file or per class. I'll change tomorrow.

Reply:
Let's go with a class property. I think if we were to implement such a dynamic caching logic there's a good chance the checks you mention would live inside the ZipFS rather than the backend.

Reply:
Pushed